.github/workflows/format-checks.yml

```diff
@@ -0,0 +1,28 @@
+name: Format checks
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+jobs:
+  check-code-format:
+    name: Check code format (C/C++)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install packages
+        run: sudo apt-get install -y clang-format-10
+      - name: Configure clang-format
+        run: >
+          sudo update-alternatives --remove-all clang-format &&
+          sudo update-alternatives --install /usr/bin/clang-format clang-format
+          /usr/bin/clang-format-10 100
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Check code format
+        run: >
+          clang-format --version &&
+          find .
+          -type f -iregex '^.*\.\(c\|cpp\|h\)$' -and -not
+          -iregex '^\.\/\(externals\|build\).*$' -and -not
+          -iregex '^.*\(hlsl\|glsl\|ps\|vs\)\.h$' |
```

Comment:
I'd like to eventually just disable clang format in these files directly instead of having to filter them. We can remove this once that's done

Reply:
Could we just have clang-format actually just FORMAT those files?

Reply:
That's true we could. I assumed since they are generated we didn't want to but I can format them

Reply:
I'll put up a PR to format them and adjust this to not exclude them

Reply:
The hlsl/glsl/vs/ps files are NOT autogenerated (I think that is true of all of them).

Reply:
Oh wait
sorry
Yes please avoid those.  I thought you were talking about the actual hlsl files not the generated headers.